CMakeLists.txt

```diff
@@ -9,45 +9,52 @@ cmake_minimum_required(VERSION 3.3)
 # In order to bootstrap the runtime library we need to skip
 # CMake's Fortran tests
 SET(CMAKE_Fortran_COMPILER_WORKS 1)
+SET(CMAKE_Fortran_ABI_COMPILED 0)
+SET(CMAKE_Fortran_COMPILER_SUPPORTS_F90 1)
+SET(CMAKE_Fortran_MODDIR_FLAG "-J ")
```

Comment:
Do you run CMake with `-DCMAKE_Fortran_COMPILER_ID=Flang`? That should set `CMAKE_Fortran_MODDIR_FLAG` correctly.